### PR TITLE
add Cognito user sdk integration test for xamarin devices[NEW]

### PR DESCRIPTION
**Pull request by sunchunqiang**
## Description

This change is to add cognito user pool high level features integration test for xamarin.

## Motivation and Context

High level features of Cognito user pool sdk are implemented for dotnet Core in this solution. As the implementation is based on dotnet standard, it will also support Xamarin. There are only integration test for dotnet core, this change will add integration test for xamarin.

## Testing

- Run Android test application.
- Run iOS test application.
- Run UWP test application.
- Run .Net core test application for a recursive test.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have read the **README** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-sdk-net/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: extensions/test/CognitoAuthentication.IntegrationTests/MfaAuthenticationTests.cs [hunk 1]
```diff
@@ -110,7 +106,8 @@ public MfaAuthenticationTests()
             verifiedAttributes.Add(CognitoConstants.UserAttrPhoneNumber);
 
             //Create Role for MFA
-            using (var managementClient = new AmazonIdentityManagementServiceClient())
+            using (var managementClient = GetAmazonIdentityManagementServiceClient() 
+                )
```
Comment (sstevenkang) [inline extensions/test/CognitoAuthentication.IntegrationTests/MfaAuthenticationTests.cs]: why is this in a new line?

File: extensions/test/CognitoAuthentication.IntegrationTests.Xamarin/CognitoAuthentication.IntegrationTests.Xamarin/_xamarin/BaseAuthenticationTestClass.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+{
+    /// <summary>
+    /// Base class to be used for authentication integrations tests
+    /// Allows for child classes to create, sign up, or confirm users
+    /// </summary>
+    public partial class BaseAuthenticationTestClass
+    {
+        protected RegionEndpoint clientRegion = Settings.RegionEndpoint; 
+
+        public AmazonCognitoIdentityProviderClient GetAmazonCognitoIdentityProviderClient()
```
Comment (sstevenkang) [inline extensions/test/CognitoAuthentication.IntegrationTests.Xamarin/CognitoAuthentication.IntegrationTests.Xamarin/_xamarin/BaseAuthenticationTestClass.cs]: could you fix the whitespace changes?  

setting the default to convert tabs to 4 spaces should help.  See this [stackoverflow](https://stackoverflow.com/questions/14167033/visual-studio-replace-tab-with-4-spaces).

File: extensions/test/CognitoAuthentication.IntegrationTests/_coreclr/BaseAuthenticationTestClass.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
+﻿/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (sstevenkang) [inline extensions/test/CognitoAuthentication.IntegrationTests/_coreclr/BaseAuthenticationTestClass.cs]: please update all copyright notices to 2018.

also, doesn't this file require update to [CognitoAuthentication.IntegrationTests.CoreCLR.csproj](https://github.com/aws/aws-sdk-net/blob/master/extensions/test/CognitoAuthentication.IntegrationTests/CognitoAuthentication.IntegrationTests.CoreCLR.csproj) ?   As it is the .csproj is picking up both copies of the BaseAuthenticationTestClass.cs, no?

File: extensions/test/CognitoAuthentication.IntegrationTests/MfaAuthenticationTests.cs (hunk 1 above)
Comment (sunchunqiang) [reply extensions/test/CognitoAuthentication.IntegrationTests/MfaAuthenticationTests.cs]: As .Net Core and .Net Xamarin use different constructor function of AmazonIdentityManagementServiceClient, we move the constructor statements to a GetAmazonIdentityManagementServiceClient and implement it in the partial class BaseAuthenticationTestClass

File: extensions/test/CognitoAuthentication.IntegrationTests/_coreclr/BaseAuthenticationTestClass.cs (hunk 3 above)
Comment (sunchunqiang) [reply extensions/test/CognitoAuthentication.IntegrationTests/_coreclr/BaseAuthenticationTestClass.cs]: We don't have to, as the source _coreclr\BaseAuthenticationTestClass.cs will be included to project CognitoAuthentication.IntegrationTests.CoreCLR.csproj automatically. CognitoAuthentication.IntegrationTests.CoreCLR.csproj will include all files under the project folder and its sub folders to the project

File: extensions/test/CognitoAuthentication.IntegrationTests/MfaAuthenticationTests.cs (hunk 1 above)
Comment (sstevenkang) [reply extensions/test/CognitoAuthentication.IntegrationTests/MfaAuthenticationTests.cs]: I know why you made the change :)

Just asking why the `)` is in a new line.

File: extensions/test/CognitoAuthentication.IntegrationTests.Xamarin/CognitoAuthentication.IntegrationTests.Xamarin/_xamarin/BaseAuthenticationTestClass.cs (hunk 2 above)
Comment (sunchunqiang) [reply extensions/test/CognitoAuthentication.IntegrationTests.Xamarin/CognitoAuthentication.IntegrationTests.Xamarin/_xamarin/BaseAuthenticationTestClass.cs]: Submitted a change to fix the issue

File: extensions/test/CognitoAuthentication.IntegrationTests/MfaAuthenticationTests.cs (hunk 1 above)
Comment (sunchunqiang) [reply extensions/test/CognitoAuthentication.IntegrationTests/MfaAuthenticationTests.cs]: Not intentional. Already fixed it

File: extensions/test/CognitoAuthentication.IntegrationTests.Xamarin/CognitoAuthentication.IntegrationTests.Xamarin/_xamarin/BaseAuthenticationTestClass.cs (hunk 2 above)
Comment (sstevenkang) [reply extensions/test/CognitoAuthentication.IntegrationTests.Xamarin/CognitoAuthentication.IntegrationTests.Xamarin/_xamarin/BaseAuthenticationTestClass.cs]: I still see a lot of indentation issues esp. lines 44~55.

File: extensions/test/CognitoAuthentication.IntegrationTests.Xamarin/CognitoAuthentication.IntegrationTests.Xamarin/Settings.cs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            {
+                var settingsStream = GetResourceStream(settingsResourcePartialName);
+                string json;
+                using (var reader = new StreamReader(settingsStream))
+                {
+                    json = reader.ReadToEnd();
+                }
+                storedSettings = JsonMapper.ToObject<StoredSettings>(json);
+            }
+            catch(Exception e)
```
Comment (assyadh) [inline extensions/test/CognitoAuthentication.IntegrationTests.Xamarin/CognitoAuthentication.IntegrationTests.Xamarin/Settings.cs]: whitespace + Should we catch JsonMapper typed exceptions first? What happens if the mapping doesn't work?

File: extensions/test/CognitoAuthentication.IntegrationTests.Xamarin/CognitoAuthentication.IntegrationTests.Xamarin/_xamarin/BaseAuthenticationTestClass.cs (hunk 2 above)
Comment (sunchunqiang) [reply extensions/test/CognitoAuthentication.IntegrationTests.Xamarin/CognitoAuthentication.IntegrationTests.Xamarin/_xamarin/BaseAuthenticationTestClass.cs]: there are tabs in BaseAuthenticationTestClass.cs, App.xaml.cs and MainPage.xaml.cs.  subbmiyed a change fix the issue

File: extensions/test/CognitoAuthentication.IntegrationTests.Xamarin/CognitoAuthentication.IntegrationTests.Xamarin/Settings.cs (hunk 4 above)
Comment (sunchunqiang) [reply extensions/test/CognitoAuthentication.IntegrationTests.Xamarin/CognitoAuthentication.IntegrationTests.Xamarin/Settings.cs]: add handler to JsonException

**Comment by sstevenkang** [comment]
We are moving this library, so let's open this PR in the new repo.

**Pull request closed without merging**